### ISSUE TITLE
Remote WAN access/WAN IP detection missing

### ISSUE THREAD

**Issue by aenima99x** [jellyfin/jellyfin#236]
It looks like the Remote WAN IP detection has been removed and only the LAN IP is shown.
Would it be possible to get the WAN IP detection added back in? 192

**Comment by joshuaboniface** [comment]
We removed this due to it being a visual-only feature that required communication with Emby's external services. Aside from quickly viewing your WAN address (which can be found by other methods), what is the usecase? I'd be open to adding it back in using another service if it's worthwhile.

**Comment by aenima99x** [comment]
It has to do with the Emby for Kodi client looking for it and throwing errors when it can't find it. I put a thread on reddit here - https://old.reddit.com/r/jellyfin/comments/a7ohrs/remote_wan_accesswan_ip_detection/
Let me know if you want more info, additional, logs, etc

**Comment by joshuaboniface** [comment]
Nope that seems fair, we'll work to reintegrate that.

**Comment by aenima99x** [comment]
Perfect, thanks!

**Comment by JustAMan** [comment]
One of easiest solutions is to make a simple DNS query to Google DNS server, like this:
```
$ dig TXT +short o-o.myaddr.l.google.com @ns1.google.com
"1.2.3.4"
```

**Comment by sparky8251** [comment]
For me, that dig command returns my v6 address which could be properly obtained without going out to the internet. Plus, I imagine many folks prefer v4 when routing these days (not to mention the lack of v6 support from ISPs).

I also know that a number of folks don't trust google very much so maybe an alternative should be found just to assuage those fears and concerns.

**Comment by sparky8251** [comment]
Looking around briefly, there does not appear to be any one easy answer. 

Sites like https://ipapi.com/ offer 10k/month free API requests. We could have a user sign up there and input an API key that we then use to obtain the WAN address reliably. This isn't terribly desirable as it requires another account and additional setup.

There are DNS lookup options, requiring a user to own a domain and have the Jellyfin server live on it. Again, not that desirable as not only will it require an end user to spend money on a domain they can add a and PTR records to, its not uncommon to have different DNS records on the LAN compared to the WAN. Unsure how Jellyfin would take it if it received an internal v4 address for the remote IP on some devices but not others.

We can setup an official Jellyfin server that responds to IP address lookup requests on one of the Jellyfin domains. This involves phoning home, something the team has been against. It also adds operational costs to Jellyfin that will only grow with popularity. Not exactly the best solution long term even if the phone home bit is OK.

I would suggest that maybe we for the Emby For Kodi addon and determine why it queries the server for the remote IP. I feel that it is likely part of Emby's auto configuration, allowing it to connect to the server via its remote IP with nothing more than your official Emby account.

This has 2 problems thinking about it. 
1) Jellyfin seems to have no plans of re-implementing Emby Connect meaning that this auto-detection and auto-configuration path is worthless. 
2) Not everyone has NAT reflection enabled on their router (I know I don't, security issue), meaning that it would be impossible to connect to the server locally using its remote IP address for some during the auto-configuration. Meaning even if the Emby Connect stuff was redone to at least handle this kind of auto-config, it would have to hand multiple addresses to the Kodi addon and then Kodi would have to pick the local address some of the time and the remote other times, adding complexity to the addon.

Maybe we should tackle this in the Kodi addon by removing the remote IP and Emby Connect options, forcing you to enter an IP address or DNS address that resolves to the Jellyfin server?

**Comment by nvllsvm** [comment]
I'm fond of the icanhazip sites.

https://major.io/icanhazip-com-faq

**Comment by joshuaboniface** [comment]
Excellent write-up @sparky8251 thanks for that!

We are indeed against the idea of hosting a centralized solution for this, both for privacy (that service will always know your public IP then and of course that you're running a Jellyfin instance) and for cost/scalability as you mention. You're correct that we're not planning to implement Emby Connect-type features either, since these are fundamentally build around the idea of a central, non-user-controlled authority.

Ignoring the Kodi add-on issue, the best solution IMO is a solid guide on setting up a dynamic DNS and port forwarding system, something on the Wiki, that would make the process as easy as possible for novice users to set up. I personally perfer `ipinfo.io` but `icanhazip` works too ;-)

The Kodi addon is it's own issue, as we mention I don't see a really good reason why it needs it, so removing the requirement might be the better move, but we'll have to review that in particular.

**Comment by sparky8251** [comment]
Cool! 

http://ipv4.icanhazip.com/
and 
http://ipv6.icanhazip.com/

Return plain old addresses with no obvious rate or daily limits. This defiantly fits the bill for a fix.

I'd think that perhaps we should bandaid it with an IP lookup for now so that the existing addon works then revisit the Kodi addon later. I really cant see how it'd be useful to Jellyfin as the project stands now.

**Comment by aenima99x** [comment]
I'm in agreement that any type of phone-home and/or hosted solution is not ideal and opverkill for this.
I think the icanhazip.com lookup is a good temporary fix until the Kodi addon can be updated to not need the wan IP and/or replaced with a jellyfin client.

**Comment by anthonylavado** [comment]
I think the `icanhazip` route is the way to go. If anyone wants to draft a PR...

**Comment by jonwest** [comment]
This PR fixes the Emby Kodi plugin for me (https://github.com/jellyfin/jellyfin-kodi/pull/1).  I'm not sure that it's a clean long-term fix, but it's working for the time being.

Interestingly, I had the Emby plugin installed on another device that was previously connected to my Emby (pre-Jellyfin) server, and it continues to work without issue.  When I install the plugin onto a _new_ device, I get the irrecoverable errors mentioned in the PR.

Also not really sure how to push this out, as I don't think there's currently a "Jellyfin Kodi Repository" where the plugin can be installed, no?  This is my first contribution into the Kodi/Emby/Jellyfin ecosystem and just fixed a problem I was having locally.

**Comment by joshuaboniface** [comment]
Closed by #305.

**Comment by anthonylavado** [comment]
Ref #679
